scipy/interpolate/src/_fitpackmodule.c

```diff
@@ -191,7 +191,7 @@ fitpack_bispev(PyObject *dummy, PyObject *args)
     ny = ap_ty->dimensions[0];
     mx = ap_x->dimensions[0];
     my = ap_y->dimensions[0];
-    mxy = mx*my;
+    mxy = (npy_uint64)mx * (npy_uint64)my;
     if (my != 0 && mxy/my != mx) {
```

Comment:
You can do this check without any overflow occurring:
```suggestion
    if (my != 0 && INTPTR_MAX/my < mx) {
```

That will work even on 32-bit platforms where the assignment to `mxy` will convert to 32-bit int.

Reply:
I will amend to this, in order to make the pipeline test work, but all previous solution worked in integer_overflow test from test_fitpack.py

Reply:
In that case test for the integer_overflow will fail due to there would be no overflow on 64bit system.

Reply:
But I guess it will fail in test `integer_overflow` from test_fitpack.py for 64bit machine due to none overflow existed.

Reply:
@peterbell10: right, indeed overflowing uint64 -> intptr_t cast would again be undefined, so probably that is the right answer